### PR TITLE
[DirectX][Docs] Add DXILIntrinsicExpansion  Pass to DXILArchitecture.rst

### LINKED ISSUE #84839: [HLSL] Add documentation for new instruction expansion pass
@farzonl  - can you fill in some more information here please?

### PR DESCRIPTION
**Pull request by farzonl**
Completes #84839

### CODE REVIEW

**Comment by damyanp** [comment]
Is there anything we can do with the PR title so that the key bit of information (ie that it is about instruction expansion) doesn't fall off the title?

File: llvm/docs/DirectX/DXILArchitecture.rst [hunk 1]
```diff
@@ -61,6 +61,19 @@ on the utilities described in "Common Code" above in order to share
 logic with both the DirectX backend and with Clang's codegen of HLSL
 support as much as possible.
 
+The DirectX Intrinsic Expansion Pass
+====================================
+There are intrinsics that don't map directly to DXIL Ops. In some cases
+an intrinsic needs to be expanded to a set of LLVM IR instructions. In
+other cases an intrinsic needs modifications to the arguments or return
+values of a DXIL Op. This pass (`DXILIntrinsicExpansion.cpp`) handles all
```
Comment (damyanp) [inline llvm/docs/DirectX/DXILArchitecture.rst:69]:
```suggestion
values of a DXIL Op. The `DXILIntrinsicExpansion` pass handles all 
```

I think this is more consistent with the other pass descriptions above.

File: llvm/docs/DirectX/DXILArchitecture.rst [hunk 2]
```diff
@@ -61,6 +61,19 @@ on the utilities described in "Common Code" above in order to share
...
+The DirectX Intrinsic Expansion Pass
+====================================
+There are intrinsics that don't map directly to DXIL Ops. In some cases
+an intrinsic needs to be expanded to a set of LLVM IR instructions. In
+other cases an intrinsic needs modifications to the arguments or return
+values of a DXIL Op. This pass (`DXILIntrinsicExpansion.cpp`) handles all 
+the cases where our intrinsics don't have a one to one mapping. You should 
+also consider using this pass when the expansion is specific to DXIL so 
+as to keep implementation details out of CodeGen. Finally, there is an 
+expectation that we maintain vectors through this pass. Therefore, best
```
Comment (damyanp) [inline llvm/docs/DirectX/DXILArchitecture.rst:73]:
```suggestion
the cases where our intrinsics don't have a one to one mapping. This pass 
may also be used when the expansion is specific to DXIL so 
as to keep implementation details out of CodeGen. Finally, there is an 
expectation that we maintain vectors through this pass. Therefore, best 
```

Suggestion is to try and keep similar voice to rest of doc.

Comment (damyanp) [inline llvm/docs/DirectX/DXILArchitecture.rst:73]: > vectors

Is "vectors" in this context unambiguous enough for the intended audience?  Would something like "vector types" be clearer?

**Review by damyanp — APPROVED** [review]

**Review by llvm-beanz — APPROVED** [review]

**Pull request merged**